harvester_e2e_tests/integration/test_images.py

```diff
@@ -1,47 +1,385 @@
+
+
+def create_with_invalid_url(api_client, unique_name):
+    code, data = api_client.images.create_by_url(unique_name, f"https://{unique_name}.img")
+
+    assert 201 == code, (code, data)
+
+    endtime = datetime.now() + timedelta(minutes=3)
     while endtime > datetime.now():
         code, data = api_client.images.get(unique_name)
         image_conds = data.get('status', {}).get('conditions', [])
         if len(image_conds) > 0:
             break
-        sleep(sleep_timeout)
+        sleep(3)
+
+    assert len(image_conds) == 1, f"Got unexpected image conditions!\n{data}"
+    assert "Initialized" == image_conds[0].get("type")
+    assert "False" == image_conds[0].get("status")
+    assert "no such host" in image_conds[0].get("message")
 
-    assert "Initialized" == image_conds[-1].get("type")
-    assert "True" == image_conds[-1].get("status")
     api_client.images.delete(unique_name)
+
+
+@pytest.mark.p0
+class TestBackendImages:
+
+    @pytest.mark.p0
+    @pytest.mark.dependency(name="create_image_from_volume")
+    def test_create_image_from_volume(self, api_client, unique_name,
+                                      export_storage_class, wait_timeout):
+        """
+        Test create image from volume
+
+        Steps:
+        1. Create a volume "test-volume" in Volumes page
+        2. Export the volume to image "export-image"
+        3. Check the image "export-image" exists
+        4. Cleanup image "export-image" on Images page
+        5. Cleanup volume "test-volume" on Volumes page
+        """
+
+        spec = api_client.volumes.Spec(1)
+        code, data = api_client.volumes.create(unique_name, spec)
+
+        assert 201 == code, (code, data)
+
+        code, data = api_client.volumes.get(unique_name)
+
+        # Check volume ready
+        endtime = datetime.now() + timedelta(seconds=wait_timeout)
+        while endtime > datetime.now():
+            code, data = api_client.volumes.get(unique_name)
+            if data["status"]["phase"] == "Bound":
+                break
+            sleep(5)
+        else:
+            raise AssertionError(
+                f"Failed to delete volume {unique_name} bound in {wait_timeout} timed out\n"
+                f"Still got {code} with {data}"
+            )
+
+        get_volume(api_client, unique_name)
+
+        api_client.volumes.export(unique_name, unique_name, export_storage_class)
+
+        image_id = get_export(api_client, unique_name, wait_timeout)
+
+        delete_volume(api_client, unique_name, wait_timeout)
+        delete_image(api_client, image_id, wait_timeout)
+
+    @pytest.mark.p0
+    @pytest.mark.dependency(name="create_image_url")
+    @pytest.mark.parametrize("image_name, image_url",
+                             [("qcow2-", qcow2_base_url + qcow2_image_name),
+                              ("iso-", iso_base_url + iso_image_name)])
+    def test_create_image_url(self, image_name, image_url, unique_name, api_client, wait_timeout):
```

Comment:
@lanfon72 let's keep this to a future refactor, thanks!

Reply:
Appreciated for the review and suggestion. Since if we update the shared image fixtures  by using the request options and config.yaml content. We may need to reorganize the structure in related test cases. 
If possible, can we keep this into future refactor. I will create another PR to optimize it. many thanks.